### ISSUE TITLE
508/UX - Select a model "menu" is not keyboard accessible

### ISSUE THREAD

**Issue by iamjolly** [GSA-TTS/10x-ai-sandbox#502]
## Problem Description

When using the keyboard to access and choose available models, there is no way to select a different model using the `tab` or `arrow` keys with keyboard alone. 

Each choice appears to be coded correctly within a `button` element, and they all should be technically focusable... however, tabbing closes the menu and dumps keyboard focus out to the end of the DOM. The code for these buttons are inserted by JS into the DOM near the end of the document, away from the `button` that opens the menu.

There is also a `div role="menu"` containing the choices when expanded, and it appears to be both incorrectly used (according to [the WAI-ARIA menu spec](https://www.w3.org/WAI/ARIA/apg/patterns/menubar/)) and unnecessary for this particular use. 

## Screenshot from dev environment

<img width="1050" alt="screenshot showing the available models menu opened with notes stating that the menu is inaccessible via keyboard" src="https://github.com/user-attachments/assets/0255065d-4cad-45f4-8f91-d080af9fc76e" />
